modules/nw-egress-router-redirect-mode-ovn.adoc

```diff
@@ -0,0 +1,131 @@
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create a network attachment definition.
+
+. Create an egress router pod.
+
+. To ensure that other pods can find the IP address of the egress router pod, create a service that uses the egress router pod, as in the following example:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: egress-1
+spec:
+  ports:
+  - name: database
+    protocol: TCP
+    port: 3306
+  type: ClusterIP
+  selector:
+    name: egress-router-pod
+----
++
+After you create the service, your pods can connect to the service. Their connections are redirected to the corresponding ports on the destination IP address and the connections originate from the reserved source IP address.
+
+.Verification steps
+
+To verify that the egress router pod started and the secondary network interface is attached, complete the following procedure:
+
+. View the events for the egress router pod:
++
+[source,terminal]
+----
+$ oc get events --field-selector involvedObject.name=egress-router-pod
+----
++
+If the pod references the network attachment definition, the previous command returns output that is similar to the following:
++
+[source,text]
+----
+LAST SEEN   TYPE     REASON           OBJECT                  MESSAGE
+5m4s        Normal   Scheduled        pod/egress-router-pod   Successfully assigned default/egress-router-pod to ci-ln-9x2bnsk-f76d1-j2v6g-worker-c-24g65
+5m3s        Normal   AddedInterface   pod/egress-router-pod   Add eth0 [10.129.2.31/23]
+5m3s        Normal   AddedInterface   pod/egress-router-pod   Add net1 [192.168.12.99/24] from default/egress-router-redirect
+----
+
+. Optional: View the routing table for the egress router pod.
+// Terminology from support-collecting-network-trace.adoc
+.. Enter into a debug session on the node for the egress router pod:
++
+[source,terminal]
+----
+$ POD_NODENAME=$(oc get pod egress-router-pod -o jsonpath="{.spec.nodeName}")
+$ oc debug node/$POD_NODENAME
+----
+
+.. Set `/host` as the root directory within the debug shell. The debug pod mounts the host's root file system in `/host` within the pod. By changing the root directory to `/host`, you can run binaries contained in the host's executable paths:
++
+[source,terminal]
+----
+# chroot /host
+----
+
+.. From within the `chroot` environment console, get the container ID:
++
+[source,terminal]
+----
+# crictl ps --name egress-router-redirect | awk '{print $1}'
+----
++
+.Example output
+[source,terminal]
+----
+CONTAINER
+bac9fae69ddb6
+----
+
+.. Determine the container's process ID. In this example, the container ID is `bac9fae69ddb6`:
++
+[source,terminal]
+----
+# crictl inspect -o yaml bac9fae69ddb6 | grep 'pid:' | awk '{print $2}'
+----
++
+.Example output
+[source,terminal]
+----
+68857
+----
+
+.. Enter the network namespace of the container:
++
+[source,terminal]
+----
+# nsenter -n -t 68857
+----
+
+.. Display the routing table:
++
+[source,terminal]
+----
+# ip route
```

Comment:
@danielmellado Do we also need check iptables rules here by running below cmd? 
[root@ip-10-0-134-247 /]# iptables-save -t nat
# Generated by iptables-save v1.8.4 on Fri Dec 11 19:08:50 2020
*nat
:PREROUTING ACCEPT [0:0]
:INPUT ACCEPT [0:0]
:POSTROUTING ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
-A PREROUTING -i eth0 -j DNAT --to-destination 172.217.7.142
-A POSTROUTING -o net1 -j SNAT --to-source 10.0.134.248
COMMIT
# Completed on Fri Dec 11 19:08:50 2020
# Warning: iptables-legacy tables present, use iptables-legacy-save to see them

Reply:
@danielmellado May use "ip netns exec testing iptables -t nat -L" from https://github.com/openshift/egress-router-cni/blob/5f6271341c454d02c63abe40f3c948c2b5b58247/docs/content/post/developer-documentation.md

Reply:
it's basically up to the user, both commands would be equivalent ;)

Reply:
/LGTM